crates/holochain/src/conductor/dna_store.rs

```diff
@@ -1,54 +1,54 @@
-use holochain_types::prelude::*;
+use holochain_types::{prelude::*, share::RwShare};
 use holochain_zome_types::entry_def::EntryDef;
 use std::collections::HashMap;
 use tracing::*;
 
 /// Placeholder for real dna store
 #[derive(Default, Debug)]
-pub struct RealDnaStore {
+pub struct DnaStore {
     dnas: HashMap<DnaHash, DnaFile>,
     entry_defs: HashMap<EntryDefBufferKey, EntryDef>,
 }
 
-impl DnaStore for RealDnaStore {
+impl DnaStore {
     #[instrument]
-    fn add_dna(&mut self, dna: DnaFile) {
+    pub fn add_dna(&mut self, dna: DnaFile) {
         self.dnas.insert(dna.dna_hash().clone(), dna);
     }
-    fn add_dnas<T: IntoIterator<Item = (DnaHash, DnaFile)> + 'static>(&mut self, dnas: T) {
+    pub fn add_dnas<T: IntoIterator<Item = (DnaHash, DnaFile)> + 'static>(&mut self, dnas: T) {
         self.dnas.extend(dnas);
     }
     #[instrument]
-    fn list(&self) -> Vec<DnaHash> {
+    pub fn list(&self) -> Vec<DnaHash> {
         self.dnas.keys().cloned().collect()
     }
     #[instrument]
-    fn get_dna_def(&self, hash: &DnaHash) -> Option<DnaDef> {
+    pub fn get_dna_def(&self, hash: &DnaHash) -> Option<DnaDef> {
         self.dnas.get(hash).map(|d| d.dna_def()).cloned()
     }
     #[instrument]
-    fn get_dna_file(&self, hash: &DnaHash) -> Option<DnaFile> {
+    pub fn get_dna_file(&self, hash: &DnaHash) -> Option<DnaFile> {
         self.dnas.get(hash).cloned()
     }
-    fn add_entry_def(&mut self, k: EntryDefBufferKey, entry_def: EntryDef) {
+    pub fn add_entry_def(&mut self, k: EntryDefBufferKey, entry_def: EntryDef) {
         self.entry_defs.insert(k, entry_def);
     }
-    fn add_entry_defs<T: IntoIterator<Item = (EntryDefBufferKey, EntryDef)> + 'static>(
+    pub fn add_entry_defs<T: IntoIterator<Item = (EntryDefBufferKey, EntryDef)> + 'static>(
         &mut self,
         entry_defs: T,
     ) {
         self.entry_defs.extend(entry_defs);
     }
-    fn get_entry_def(&self, k: &EntryDefBufferKey) -> Option<EntryDef> {
+    pub fn get_entry_def(&self, k: &EntryDefBufferKey) -> Option<EntryDef> {
         self.entry_defs.get(k).cloned()
     }
 }
 
-impl RealDnaStore {
-    pub fn new() -> Self {
-        RealDnaStore {
+impl DnaStore {
```

Comment:
we don't need two identical impl blocks next to each other, these can be merged now.